Fix custom style issues

firmusoft opened the pull request:
1. Fix: In collaborative mode Mismatch transaction error on apply custom style
2. Fix: Numbering is applied to a paragraph which is not having styles on delete/backspace.
3. Validation for selected Nodes.
4. css counter variable's name changed(Lengthy line not allowed by Sonar Scanner).

Seybi-ea committed:
Fix: Numbering is applied to paragraph which is not having custom style on delete.

Commit:
FIX: Lengthy line not allowed by Sonar Scanner.

Commit:
FIX for IRAD-1264.

Commit:
Validate styleProp.styles

Commit:
Validate selectedNode

Seybi-ea committed:
Fix:Numbering issues(Chnaged Counter names)

firmusoft committed:
Merge branch 'main' into fix_customstyle_issues

Seybi-ea committed:
Licit version changed

melgish commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@modusoperandi/licit",
-  "version": "0.0.24",
+  "version": "0.0.25",
```
**DO NOT** modify the version of package.json without making the exact same change in package-lock.json.  Better to leave unchanged until release is created.

melgish commented on src/CustomStyleCommand.js:
```diff
@@ -657,55 +653,64 @@ function applyStyleEx(
+    // [FS] IRAD-1074 2020-10-22
+    // Issue fix on not removing center alignment when switch style with center
+    // alignment to style with left alignment
+    newattrs['align'] = null;
```
Using string literals to access object members is discouraged.   Bypassing Flow makes the project harder to maintain.

melgish commented on src/CustomStyleCommand.js:
```diff
@@ -657,55 +653,64 @@ function applyStyleEx(
+          .paragraphSpacingAfter
+          ? styleProp.styles.paragraphSpacingAfter
+          : null;
+        newattrs['paragraphSpacingBefore'] = styleProp.styles
```
Don't use ternary syntax in places where null coalescence can serve better.

newattrs.paragraphSpacingAfter = styleProp.styles.paragraphSpacingAfter || null;
newattrs.paragraphSpacingBefore = styleProp.styles.paragraphSpacingBefore || null;

melgish commented on src/CustomStyleCommand.js:
```diff
@@ -808,7 +813,7 @@ function hasMismatchHeirarchy(
     if (levelDiff > 0) {
       if (selectedNodes.length === 1) {
-        nodesBeforeSelection.reverse();
+        // nodesBeforeSelection.reverse();
```
Don't leave commented out code.  Keeping history is what GitHub is for.
Don't leave empty conditions.

melgish commented on src/ui/ImageNodeView.js:
```diff
@@ -289,23 +289,25 @@ class ImageViewBody extends React.PureComponent<any, any> {
   };
 
   _onKeyDown = (e: any): void => {
     console.log(e.keyCode);
```
Don't spam console with keystrokes.

melgish commented on src/ui/mathquill-editor/MathQuillEditor.js:
```diff
@@ -10,8 +10,8 @@ import cx from 'classnames';
 // Now loaded locally, so that it work in closed network as well.
-import type { MathQuillEditorSymbol } from './MathQuillEditorSymbols';
-import { MathQuill } from './mathquill-import-kludge';
+import type {MathQuillEditorSymbol} from './MathQuillEditorSymbols';
```
Do not commit 'whitespace only' changes

melgish reviewed (changes requested):
In addition to other comments, running `npm run flow` reports 148 additional errors that need to be reviewed.

Seybi-ea committed:
1. Fix: Custom styles not loading on Readonly mode.
2. Resolved the review comments except flow validation error.

Seybi-ea committed:
1. Added new line after table position on table drop.
2. Fix: Extra  empty element adding on applying custom style with Level.
3. Fix: Unable to drop table after deleting whole content from the document.

melgish commented:
There are now 150 flow errors instead of 148.
